packages/dd-trace/src/ritm.js

```diff
@@ -69,8 +70,19 @@ function Hook (modules, options, onrequire) {
     if (!patched) {
       patching[filename] = true
     }
+    const payload = {
+      filename,
+      request
+    }
 
+    if (moduleLoadStartChannel.hasSubscribers) {
```

Comment:
This will be called multiple times for circular dependencies. Can you add separate code paths for when `patched` is true and false? When it's true then all that's needed is to `return origRequire.apply(this, arguments)` and otherwise all of the code would execute.